jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/SecureRequestCustomizer.java

```diff
@@ -29,63 +28,31 @@
-import org.eclipse.jetty.io.ssl.SslConnection.SslEndPoint;
-import org.eclipse.jetty.util.StringUtil;
 import org.eclipse.jetty.util.annotation.Name;
 import org.eclipse.jetty.util.ssl.SslContextFactory;
 import org.eclipse.jetty.util.ssl.X509;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
- * <p>Customizer that extracts the attribute from an {@link SSLContext}
- * and sets them on the request with {@link Request#setAttribute(String, Object)}
- * according to Servlet Specification Requirements.</p>
+ * <p>Customizer that extracts the attribute of an {@link SSLContext}
+ * and makes them available via {@link Request#getAttribute(String)}
+ * using the names:
+ * <ul>
+ *     <li>{@link #SSL_SESSION_ATTRIBUTE} for the {@link SSLSession} itself</li>
+ *     <li>{@link #SSL_SESSION_DATA_ATTRIBUTE} for {@link SSLSession} metadata</li>
+ *     <li>{@link #X509_ATTRIBUTE} for the local certificate as a {@link X509} instance</li>
+ * </ul>
+ * @see EndPoint.SslSessionData
  */
 public class SecureRequestCustomizer implements HttpConfiguration.Customizer
 {
-    /**
-     * <p>The request attribute name to use to obtain the cipher suite name.</p>
-     */
-    public static final String CIPHER_SUITE_ATTRIBUTE = "org.eclipse.jetty.server.cipher";
-    /**
-     * <p>The request attribute name to use to obtain the key size.</p>
-     */
-    public static final String KEY_SIZE_ATTRIBUTE = "org.eclipse.jetty.server.keySize";
-    /**
-     * <p>The request attribute name to use to obtain the {@link SSLSession#getId()}.</p>
-     */
-    public static final String SSL_SESSION_ID_ATTRIBUTE = "org.eclipse.jetty.server.sslSessionId";
-    /**
-     * <p>The request attribute name to use to obtain the peer certificate
-     * chain as an array of {@link X509Certificate} objects.</p>
-     */
-    public static final String PEER_CERTIFICATES_ATTRIBUTE = "org.eclipse.jetty.server.peerCertificates";
-    /**
-     * <p>The request attribute name to use to obtain the local certificate
-     * as an {@link X509} object.</p>
-     */
     public static final String X509_ATTRIBUTE = "org.eclipse.jetty.server.x509";
-    /**
-     * <p>The default value of the request attribute name to use to obtain
-     * the {@link SSLSession} object.</p>
-     *
-     * @see #setSslSessionAttribute(String)
-     */
-    public static final String DEFAULT_SSL_SESSION_ATTRIBUTE = "org.eclipse.jetty.server.sslSession";
-    /**
-     * <p>The default value of the request attribute name to use to obtain
-     * the {@link SslSessionData} object.</p>
-     *
-     * @see #getSslSessionDataAttribute()
-     */
-    public static final String DEFAULT_SSL_SESSION_DATA_ATTRIBUTE = newSslSessionDataAttribute(DEFAULT_SSL_SESSION_ATTRIBUTE);
+    public static final String SSL_SESSION_ATTRIBUTE = "org.eclipse.jetty.server.SslSession";
+    public static final String SSL_SESSION_DATA_ATTRIBUTE = EndPoint.SslSessionData.ATTRIBUTE;
```

Comment:
We only break since 12.0.0, since before that they had "javax" or "jakarta" in their names.    Every attribute has a cost and it is far better to put related values behind a single attribute rather than repeat the mistake of the servlet spec to have multiple individual attributes.

I say we break 12.0.0 and if anybody complains we can add them back as deprecated attributes.

Reply:
I have added them back in as an optional legacy support, marked for removal.